bip-xxx.mediawiki

```diff
@@ -36,50 +36,107 @@ The centralized solution has two issues:
+
+In this case the merchant’s payment process queries the exchange to get current rate information and a destination address.   The exchange notifies the payment processor when payment is received.  The exchange converts the bitcoin according to rules setup during the configuration.
+
+The setup:
 
 * Manny (the "merchant") wants to accept Bitcoin payments on his e-commerce website.
 * Manny chooses the payment processor "PROCCO" which has a powerful plugin for his e-commerce website.
 * Manny is based in Canada and already has an account on the exchange "MYCOIN" which supports the Crypto Open Exchange Protocol.
 * Manny connects to the exchange website, and creates a new address source.
 * In the configuration screen of the address source, for each payment sent to this address source, Manny decides to keep 30% in Bitcoin and place a market sell order for the remaining 70% of the amount.
+* Manny provides "PROCCO's" webhook endpoint that so "PROCCO" is notified when payments have been made and confirmed. 
 * "MYCOIN" creates the address source, and gives the "address source URI" to the merchant. (e.g. https://example.com/addresssources/abd29ddn92)
 * Manny copies the address source URI and goes inside "PROCCO" settings, and configures his store to use this address source URI.
 
-Now a customer, Carol, wants to order a brand new phone for 0.01 BTC on Manny's store and decides to pay in Bitcoin.
+Now a customer, Carol, wants to order a brand new phone for $500 CAD on Manny's store and decides to pay in Bitcoin.
 
 * The E-Commerce website plugin requests the creation of an invoice from PROCCO. 
 * PROCCO queries the "address source URI" and retrieves the rate, the expiration of this rate and conditions.
 * PROCCO can now show the Bitcoin Payment Checkout page.
-* Carla pays.
-* PROCCO marks the payment as paid and redirects to the e-commerce website.
-* MYCOIN, under its own policy (typically after 6 confirmations), credits Manny's account of 0.01 BTC and simultaneously creates a market sell order of 0.007 BTC on behalf of Manny.
+* Carla pays.  MYCOIN sees the payment in the mempool and notifies PROCCO that a payment has been made via the webhook.
+* PROCCO updates the invoice to acknowledge that a payment has been sent.  
+* MYCOIN, under its own policy (typically after 6 confirmations), credits Manny's account of 0.01 BTC and simultaneously creates a market sell order of 0.007 BTC on behalf of Manny.   MYCOIN notifies PROCCO that the payment was confirmed and finalized via the webhook.
+* PROCCO marks the payment as paid/finalized.   Manny sees the order has been paid and processes the order as usual.
+
+===Case II: Merchant uses exchange’s LN=== 
+
+In this case, the Payment Processor requests a LN address from the exchange.  The exchange keeps the Payment Processor up-to-date on payment status and sells the funds as configured.
+
+The steps are identical to Case I except LN addresses are sent instead of Bitcoin.
```

Comment:
I think a different field should be used. The exchange could allow any of them, Carol is the one choosing whether to pay on LN or BTC depending on her wallet.

Reply:
Makes sense.  I had assumed that LN & BTC would be different endpoints at the exchange but it probably makes more sense for there to be a single endpoint that just offers both options if available.